the-collab-lab/tcl-13-smart-shopping-list
# Ss rb no duplicates

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description


- Refactoring to useContext for all data being passed from app
- Refactoring to use OnSnapShot to subscribe to db and listen for updates
- Created compare Items function in Add Items component that uses regex expression to ensure no duplicate db entries
- Removed check token component and replaced with Redirect components in the Router based on token presence 
- When the user doesn't have a token stored in `localStorage`, the Welcome screen is shown so that users can enter a token or create a new list. 
- If the user enters an existing token, clicking submit saves the token to `localStorage` and the user can access the list. The database is checked to ensure the token exists, otherwise an error is thrown and the user can try again or create a new list

## Related Issue



Fixes #5 and #6 

## Acceptance Criteria




## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After



![image](https://user-images.githubusercontent.com/17222004/96808608-2986b800-13e7-11eb-8733-d590f3d1abf3.png)

## Testing Steps / QA Criteria



Open app on localhost and create a new list. You should then be redirected to the view list component. Within view List if you add Items to the list the page should re-render with the newly added item. If the item already exists in the database you should get an alert prompt saying the item is duplicated and won'e be added to the database.

If there is no token in `localStorage`, you should be redirected to the Welcome component which allows you to enter an existing token or create a new list. If you create a new list, a new token should be stored in `localStorage`. If you enter a token the does not exist, an error should be thrown. If you enter an existing token (ex. 'creek shim pest'), it should then be appear in `localStorage` and you should have access to the list in ViewList.
